builds/e2e/templates/e2e-run.yaml

```diff
@@ -65,8 +70,6 @@ steps:
     E2E_REGISTRIES__0__PASSWORD: $(TestContainerRegistryPassword)
     E2E_ROOT_CA_PASSWORD: $(TestRootCaPassword)
     E2E_BLOB_STORE_SAS: $(TestBlobStoreSas)
-    http_proxy: $(Agent.ProxyUrl)
-    https_proxy: $(Agent.ProxyUrl)
```

Comment:
I don't know why it wasn't causing problems before, but it was causing problems for me in the non-proxy jobs. In Azure Pipelines, [macro variables without values don't coalesce to empty strings](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#macro-syntax-variables), so in non-proxy jobs the value of `https_proxy` was being set to the string literal "$(Agent.ProxyUrl)". I tried a few other options for variable syntax and couldn't get it to work. Since it turns out they aren't necessary anyway, I just removed them. I wish I could remember why I added them in the first place...